File: week02/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Main variables
    string txt = get_string("Text: ");

    int letters = 0;
    int words = 1;
    int sentences = 0;

    // Main procedure
    for (int i = 0; i < strlen(txt); i++)
    {
        if (isalpha(txt[i]))
        {
            letters++;
        }

        else if (isspace(txt[i]))
        {
            words++;
        }

        else if (txt[i] == '.' || txt[i] == '!' || txt[i] == '?')
        {
            sentences++;
        }
    }

    // Letters and sentences for index
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    // Index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Procedures of index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

File: week06/sentimental-readability/readability.py
text = input("Text: ")

# Count number of letters, words, and sentences in input text
num_letters = sum(c.isalpha() for c in text)
num_words = len(text.split())
num_sentences = len([c for c in text if c in ['.', '!', '?']])

# Calculate Coleman-Liau index
L = num_letters / num_words * 100
S = num_sentences / num_words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Print grade level based on Coleman-Liau index
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade", index)
